repository/src/interpret/push.rs

```diff
@@ -0,0 +1,113 @@
+                .to_string()[0..BRANCH_NAME_HASH_DIGITS];
+            let signature = TypedSignature::sign(
+                &(commit_hash, branch.to_owned(), timestamp as u64),
+                private_key.as_ref().unwrap(),
+            )?;
+            let signer = serde_spb::to_string(signature.signer())?.replace('\"', "\\\"");
+            let signature =
+                serde_spb::to_string(&signature.get_raw_signature())?.replace('\"', "\\\"");
+
+            raw.create_branch(branch.into(), commit_hash).await?;
+            raw.push_option(
+                remote_name.clone(),
+                branch.into(),
+                Some(format!(
+                    "{commit_hash} {branch} {timestamp} {signature} {signer}"
+                )),
+            )
+            .await?;
+            raw.delete_branch(branch.into()).await?;
+        }
+
+        for &(commit_hash, _) in &blocks {
+            let timestamp = get_timestamp();
+            let branch = &commit_hash
+                .to_hash256()
+                .aggregate(&timestamp.to_hash256())
+                .to_string()[0..BRANCH_NAME_HASH_DIGITS];
+            let signature = TypedSignature::sign(
+                &(commit_hash, branch.to_owned(), timestamp as u64),
+                private_key.as_ref().unwrap(),
+            )?;
+            let signer = serde_spb::to_string(signature.signer())?.replace('\"', "\\\"");
+            let signature =
+                serde_spb::to_string(&signature.get_raw_signature())?.replace('\"', "\\\"");
+
+            raw.create_branch(branch.into(), commit_hash).await?;
+            raw.push_option(
+                remote_name.clone(),
+                branch.into(),
+                Some(format!(
+                    "{commit_hash} {branch} {timestamp} {signature} {signer}"
+                )),
+            )
+            .await?;
+            raw.delete_branch(branch.into()).await?;
+        }
+
+        // Push fp and finalized branch.
+        let commit_hash = raw.locate_branch(FP_BRANCH_NAME.into()).await?;
+        let timestamp = get_timestamp();
+        let branch = &commit_hash
+            .to_hash256()
+            .aggregate(&timestamp.to_hash256())
+            .to_string()[0..BRANCH_NAME_HASH_DIGITS];
+        let signature = TypedSignature::sign(
+            &(commit_hash, branch.to_owned(), timestamp as u64),
+            private_key.as_ref().unwrap(),
+        )?;
+        let signer = serde_spb::to_string(signature.signer())?.replace('\"', "\\\"");
+        let signature = serde_spb::to_string(&signature.get_raw_signature())?.replace('\"', "\\\"");
+
+        raw.create_branch(branch.into(), commit_hash).await?;
+        raw.push_option(
+            remote_name.clone(),
+            branch.into(),
+            Some(format!(
+                "{commit_hash} {branch} {timestamp} {signature} {signer}"
+            )),
+        )
+        .await?;
+        raw.delete_branch(branch.into()).await?;
+    }
+    Ok(())
+}
```

Comment:
I thinks this function has too many duplicated code. Please put the target branches in a same list and iterate over it

Reply:
fixed